flink-filesystems/flink-azure-fs-hadoop/src/main/java/org/apache/flink/fs/azurefs/AbstractAzureFSFactory.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.fs.azurefs;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.core.fs.FileSystem;
+import org.apache.flink.core.fs.FileSystemFactory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Abstract factory for AzureFS. Subclasses override to specify
+ * the correct scheme (wasb / wasbs).
+ */
+public abstract class AbstractAzureFSFactory implements FileSystemFactory {
+	private static final Logger LOG = LoggerFactory.getLogger(AzureFSFactory.class);
+
+	private Configuration flinkConfig;
+
+	@Override
+	public void configure(Configuration config) {
+		flinkConfig = config;
```

Comment:
Why not use `hadoopConfig` to separate specific configurations of Azure just like other filesystem factory did?

Reply:
I'm not sure I understand. The approach I've taken is basically copying relevant Flink config entries to the hadoop config in the `AzureFileSystem` class if they match a `azure.` or `fs.azure.` prefix. Were you suggesting I base this off the S3 fs factory? or a different fs factory? (seems like the various ones are a bit different from each other on this)

Reply:
hmm, previously I thought you did not take into the `hadoopConfig`, but then I found you just follow the `MapRFsFactory`'s way to instantiate the hadoop configuration when creating the file system instead of instantiating them within `FileSystemFactory#create(URI fsUri)` before creating the file system like other file factories did.
If this works fine, it would be okay to do so.

Reply:
Yeah I was able to test this out and verify it works ok.